addons/loyalty/models/loyalty_program.py

```diff
@@ -14,6 +14,16 @@ class LoyaltyProgram(models.Model):
     _order = 'sequence'
     _rec_name = 'name'
 
+    @api.model
+    def default_get(self, fields_list):
+        defaults = super().default_get(fields_list)
+        if 'program_type' in self.env.context:
+            program_type = self.env.context['program_type']
+            program_default_values = self._program_type_default_values()
+            if program_type in program_default_values:
+                defaults.update({k: v for k, v in program_default_values[program_type]})
```

Comment:
Ok, stupid question incoming: why? :thinking: 
We want to override these whatever it was before, and every key is a needed default. 
More robust in case we change field names? But then we need to change the default values too anyway, or it won't work...
There was another mistake here, but this one? I don't get it.

Reply:
It's like the concept of 'a compute should only compute the field itself' or 'a compute should not compute stored and non stored fields together' (...), it's expected, and if you don't respect it, you will fall into the uncertain behaviors of the orm, sometimes working, sometimes not, or simply having no effect.

Reply:
:woman_facepalming: thanks =]